### PR TITLE
Adjust helm configuration to use with config-controller

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/16591

### CODE REVIEW

File: helm/kvm-operator/templates/configmap.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ data:
         address: 'http://0.0.0.0:8000'
     service:
       crd:
-        labelSelector: '{{ .Values.Installation.V1.GiantSwarm.KVMOperator.CRD.LabelSelector }}'
+        labelSelector: ''
```
Comment (corest) [inline helm/kvm-operator/templates/configmap.yaml:18]: this values is empty in all installations

File: helm/kvm-operator/templates/configmap.yaml [hunk 2]
```diff
@@ -29,28 +29,28 @@ data:
...
-        domain: 'docker.io'
-        mirrors: 'giantswarm.azurecr.io'
+        domain: '{{ .Values.registry.domain }}'
+        mirrors: '{{ range $i, $e := .Values.registry.mirrors }}{{ if $i }},{{end}}{{ $e }}{{end}}'
       tenant:
         ssh:
           ssoPublicKey: '{{ .Values.Installation.V1.Guest.SSH.SSOPublicKey }}'
         update:
-          enabled: {{ .Values.Installation.V1.Guest.Update.Enabled }}
+          enabled: true
```
Comment (corest) [inline helm/kvm-operator/templates/configmap.yaml:56]: this is true in all installations configurations

**Review by kubasobon — APPROVED** [review]

**Review by kubasobon — APPROVED** [review]

**Review by kopiczko — APPROVED** [review]

File: helm/kvm-operator/Chart.yaml [hunk 3]
```diff
@@ -6,3 +6,4 @@ version: "[[ .Version ]]"
 appVersion: "[[ .AppVersion ]]"
 annotations:
   application.giantswarm.io/team: "rocket"
+  config.giantswarm.io/version: add-kvm-operator
```
Comment (kopiczko) [inline helm/kvm-operator/Chart.yaml:9]: Reminder to revert

File: service/controller/resource/nodecontroller/create.go [hunk 4]
```diff
@@ -44,7 +44,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	}
 
 	var desiredController *nodecontroller.Controller
-	if k8sClient != nil {
+	if k8sClient != nil { //nolint
```
Comment (tfussell) [inline service/controller/resource/nodecontroller/create.go:47]: Maybe this can be removed now?

Comment (corest) [reply service/controller/resource/nodecontroller/create.go]: removed

**Pull request merged**